include/boost/spirit/repository/home/karma/nonterminal/subrule.hpp

```diff
@@ -447,6 +470,16 @@ namespace boost { namespace spirit { namespace repository { namespace karma
         }
 
         // non-const versions needed to suppress proto's %= kicking in
+#ifndef BOOST_NO_CXX11_RVALUE_REFERENCES
```

Comment:
Do you know why the `operator%=(const&,&)` and `operator%=(const&,&&)` need to exist? Can't they be removed and use the `operator%=(const& const&)` version instead? It appears that the right hand side never needs to be modified.

Reply:
You will get what was before the commit - proto's `operator%=(&&, &&)` will be used and it is not what we want.